pennylane/measurements/counts.py

```diff
@@ -301,17 +302,30 @@ def circuit(x):
 
         if self.obs is None and not isinstance(self.mv, MeasurementValue):
             # convert samples and outcomes (if using) from arrays to str for dict keys
-            samples = qml.math.array(
-                [sample for sample in samples if not qml.math.any(qml.math.isnan(sample))]
-            )
+
+            # remove nans
+            mask = qml.math.isnan(samples)
+            nsam = shape[-1]
+            if np.any(mask):
+                mask = np.logical_not(np.any(mask, axis=tuple(range(1, samples.ndim))))
+                samples = samples[mask, ...]
+
+            # convert to string
+            def convert(x):
+                return f"{x:064b}"[-nsam:]
+
+            exp2 = 2 ** np.arange(nsam - 1, -1, -1)
+            samples = np.einsum("...i,i", samples, exp2)
+            new_shape = samples.shape
             samples = qml.math.cast_like(samples, qml.math.int8(0))
-            samples = qml.math.apply_along_axis(_sample_to_str, -1, samples)
```

Comment:
please delete the unused `_sample_to_str` function above

Reply:
Hadn't realized this isn't used anymore, thanks.